src/ec/packages/ec-component-media-container/media-container.html.twig

```diff
@@ -0,0 +1,75 @@
+{% spaceless %}
+{#
+  Parameters:
+  - "alt" (string) (default: ''),
+  - "image" (string) (default: ''),
+  - "extra_classes" (string) (default: ''),
+  - "sources" (array) (default: []): format: [
+      {
+        "src" (string) (default: ''),
+        "type" (string) (default: ''),
+      },
+      ...
+    ],
+  - "tracks" (array) (default: []): format: [
+      {
+        "src" (string) (default: ''),
+        "kind" (string) (default: ''),
+        "srcLang" (string) (default: ''),
+        "label" (string) (default: ''),
+      },
+      ...
+    ],
+  - "description" (string) (default: ''),
+  - "extra_attributes" (array) (default: []): format: [
+      {
+        "name" (string) (default: ''),
+        "value" (string) (default: '')
+      },
+      ...
+    ]
+#}
+
+{# Internal properties #}
+
+{% set _css_class = 'ecl-media-container' %}
+{% set _image = image|default('') %}
+{% set _description = description|default('') %}
+{% set _alt = image|default('') %}
+{% set _extra_attributes = '' %}
+{% set _sources = sources|default({}) %}
+{% set _tracks = tracks|default({}) %}
+
+{# Internal logic - Process properties #}
+
+{% if extra_classes is defined and extra_classes is not empty %}
+  {% set _css_class = _css_class ~ ' ' ~ extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined and extra_attributes is not empty and extra_attributes is iterable %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %} 
+
+{# Print the result #}
+
+<figure class="{{ _css_class }}"{{ _extra_attributes|raw }}>
+{%- if _sources is not empty -%}
+  <video class="ecl-media-container__media" controls="controls" poster="{{ _image }}" alt="{{ _alt }}">
+  {%- for source in _sources -%}
+    <source type="{{ source.type }}" src="{{ source.src }}" />
+  {%- endfor -%}
```

Comment:
Please consider if would actually need to remove white spaces in cases such as this.

Reply:
Ah, i have to say that i'm confused, after all, regarding how to handle this, at the moment i removed probably all the dashes from control structures and it doesn't look like we would need to trim anything else but i might be totally wrong.

Reply:
I'd personally do the same. Except, please trim `{{ _description }}` as it's definitely a place where whitespaces won't be good.